core/sr-primitives/src/traits.rs

```diff
@@ -835,12 +843,15 @@ pub trait ValidateUnsigned {
 /// Opaque datatype that may be destructured into a series of raw byte slices (which represent
 /// individual keys).
 pub trait OpaqueKeys: Clone {
-	/// Return the number of encoded keys.
-	fn count() -> usize { 0 }
-	/// Get the raw bytes of key with index `i`.
-	fn get_raw(&self, i: usize) -> &[u8];
+	/// An iterator over the type IDs of keys that this holds.
+	type KeyTypeIds: IntoIterator<Item=super::KeyTypeId>;
+
+	/// Return an iterator over the key-type IDs supported by this set.
+	fn key_ids() -> Self::KeyTypeIds;
```

Comment:
Why not?

```suggestion
	fn key_ids() -> &'static [super::KeyTypeId];
```

Reply:
who says it has to be a slice? `IntoIterator` is strictly more flexible.

Reply:
No one, I just remembered some discussions about this type and trying to make key extraction `O(1)`. So, having here a static slice does not sound that strange to me.
But you are right and the compiler probably optimizes it away and it makes no difference.